workflows/flows/analyse_study_tasks/shared/study_summary.py

```diff
@@ -97,6 +100,17 @@ def generate_study_summary_for_pipeline_run(
                     non_insdc=EMG_CONFIG.amplicon_pipeline.allow_non_insdc_run_names,
                     output_prefix=pipeline_outdir.name,  # e.g. a hash of the samplesheet
                 )
+        elif analysis_type == "rawreads":
```

Comment:
This can also be refactored to reduce nesting and repetition.
The main difference between these elif blocks appears to be the generator that's used. 
As such we can have a dedicated function that accepts a generator as a parameter.